<launch>

  <node name="wpr_server_node" pkg="wpr_warehousing_monitor" type="wpr_server_node" output="screen"/>
  <node name="wpr_swarm_fake" pkg="wpr_warehousing_monitor" type="wpr_swarm_fake" output="screen"/>

  <!-- environment -->
  <!-- <group ns="environment">
    <param name="tf_prefix" value="environment"></param>
    <arg name="model" default="$(find wpr_warehousing_monitor)/urdf/environment.urdf"/>
    <arg name="gui" default="false" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  </group> -->

<group ns="wpb_mani_1">
    <param name="tf_prefix" value="wpb_mani_1"></param>
    <arg name="model" default="$(find wpb_mani_description)/urdf/wpb_mani.urdf.xacro"/>
    <arg name="gui" default="false" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  </group>

  <group ns="wpb_mani_2">
    <param name="tf_prefix" value="wpb_mani_2"></param>
    <arg name="model" default="$(find wpb_mani_description)/urdf/wpb_mani.urdf.xacro"/>
    <arg name="gui" default="false" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  </group>

  <group ns="wpb_mani_3">
    <param name="tf_prefix" value="wpb_mani_3"></param>
    <arg name="model" default="$(find wpb_mani_description)/urdf/wpb_mani.urdf.xacro"/>
    <arg name="gui" default="false" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  </group>

  <group ns="wpb_mani_4">
    <param name="tf_prefix" value="wpb_mani_4"></param>
    <arg name="model" default="$(find wpb_mani_description)/urdf/wpb_mani.urdf.xacro"/>
    <arg name="gui" default="false" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  </group>

  <group ns="wpb_mani_5">
    <param name="tf_prefix" value="wpb_mani_5"></param>
    <arg name="model" default="$(find wpb_mani_description)/urdf/wpb_mani.urdf.xacro"/>
    <arg name="gui" default="false" />
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  </group>

  <!-- Map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find wpr_warehousing_monitor)/maps/map.yaml"/>
  
  <!-- Map tools  -->
  <node pkg="waterplus_map_tools" type="wp_manager" name="wp_manager" output="screen" />

  <!-- Rviz -->
  <arg name="rvizconfig" default="$(find wpr_warehousing_monitor)/rviz/monitor.rviz" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>
